### ISSUE TITLE
Create default ("superuser") role before accepting CQL commands

### ISSUE THREAD

**Issue by nyh** [scylladb/scylladb#10481]
@kostja noticed in https://github.com/scylladb/scylla/pull/10406 that when Scylla is run with `--authenticator PasswordAuthenticator` Scylla may become responsive to CQL requests before the default (a.k.a. "superuser") user cassandra/cassandra is set up - so CQL is still not usable for a while longer, until this user is finally set up.

This a bug - CQL should not become responsive before it is expected to work correctly. Users should not experience "invalid request" errors even temporarily while a node is coming up.

Looking at the code I can't seem to find the cause of this bug (@kostja, maybe you found it?), the code in main.cc creates an auth service and calls its `start()` and wait for it to complete. This in turn (see `auth/service.cc`)  calls the role manager's `start()` function and waits for it to complete. The `standard_role_manager::start()` function (in `auth/standard_role_manager.cc`) calls `create_default_role_if_missing()` to create the default role and seems to wait on one shard for it to happen (and since main.cc waits for all shards to complete the start(), that is enough). `create_default_role_if_missing()` also seems to wait for the query that it invokes to complete.

@kostja reproduced this bug in a new test-running code (https://github.com/scylladb/scylla/pull/10406) which started Scylla and quickly sent requests when CQL was available (but apparently before the "cassandra" user become available). I haven't yet been able to reproduce it in the existing test/cql-pytest/run, although if this bug exists, test/cql-pytest/run should be vulnerable to this problem just the same.

**Comment by kostja** [comment]
I believe the root cause is standard_role_manager::start() which calls do_after_system_ready() which loops and retries.

**Comment by nyh** [comment]
In "normal" dev runs on my laptop, I see in the logs:
```
INFO  2022-05-03 18:06:14,694 [shard 0] standard_role_manager - Created default superuser role 'cassandra'.
INFO  2022-05-03 18:06:14,694 [shard 0] cql_server_controller - Starting listening for CQL clients on 127.1.222.172:9042 (unencrypted, non-shard-aware)
```
(note the order, and the same time up to the millisecond). So the test is not able to send requests before the "cassandra" role is available and this bug is not reproduced. To reproduce this problem reliably I may need to add an artificial sleep to `create_default_role_if_missing()`.

**Comment by nyh** [comment]
If I add an artificial 5-second sleep to the beginning of `create_default_role_if_missing()`, I can see the message "standard_role_manager - Created default superuser role 'cassandra'." printed only 5 seconds after the "cql_server_controller - Starting listening for CQL clients on ..." message. But the strange thing that I can't explain is that somehow, the test which creates a test keyspace waits these 5 seconds -  so I don't see an actual test failure like in #10406.

**Comment by nyh** [comment]
I easily fixed the initialization wait for `create_default_role_if_missing()`, but I'm beginning to wonder whether it is not dangerous to do so: Because the roles is a distributed table, reading from it or writing it can block (we have a timeout of a minute). This will mean that the initialization can block for a whole minute. I just wonder if this is acceptable. I don't think I have a choice, but I wonder if we have any rules against it...

**Comment by nuivall** [comment]
I think it's also relevant that password is created by separate service: `password_authenticator::create_default_if_missing` and one can't login until password is updated in the role row.

**Comment by ptrsmrn** [comment]
Questions to be answered: why do we need logic to recreate default user (usually `cassandra` user, which is also a superuser) after a node is restarted? Perhaps reaching out to someone from the field engineering would help to answer that. @ppalczak maybe?
Is this somehow connected with the password recovery procedure? Or to prevent blocking the cluster (so one cannot log into it), but this could have been superseded by the raft recovery (?)

**Comment by kostja** [comment]
I think this was a tribute to eventual consistency; now that we moved auth state to raft this code should be removed, this would fix hte bug.

**Comment by nuivall** [comment]
I think password recovery procedure should be rewritten to work via maintenance socket but we need to:
- make sure it works
- document it

**Comment by nuivall** [comment]
You're not ought to remove it, but find a better home for it :)
